### PR TITLE
Interstation Cargo Network

### PR DESCRIPTION
**Pull request by ike709**
Relevant API changes to support https://github.com/BeeStation/BeeStation-Hornet/pull/3273

Requires the comms key to be set in `private.yml`

### CODE REVIEW

**Comment by ike709** [comment]
Completely untested, by the way.

File: app/modules/api/controllers.py [hunk 1]
```diff
@@ -92,4 +92,47 @@ def page_api_bans():
...
+
+		return jsonify(d)
+
+	except Exception as E:
+		return jsonify({"error": str(E)})
+
+@bp_api.route("/cross_cargo/purchase")
+def page_api_cross_cargo_purchase():
+	comms_key = request.args.get('comms_key', type=str, default="")
+	if comms_key == cfg.PRIVATE["comms_key"]:
```
Comment (ike709) [inline app/modules/api/controllers.py:109]: I'm concerned about someone trying to bruteforce the comms key. Is there a way to ratelimit this function?

Comment (qwertyquerty) [reply app/modules/api/controllers.py:109]: Flask-Limiter package, but really this is unneeded

File: app/modules/api/controllers.py [hunk 2]
```diff
@@ -92,4 +92,47 @@ def page_api_bans():
...
+		try:
+			icn_id = request.args.get('icn_id', type=str, default="")
+			if icn_id == "":
+				return jsonify({"error": "invalid id"})
+			purchaser = request.args.get('purchaser', type=str, default="SS13")
+			server_list = page_api_cross_cargo()
```
Comment (qwertyquerty) [inline app/modules/api/controllers.py]: This doesn't work, jsonify doesn't return a dict

**Review by qwertyquerty — CHANGES_REQUESTED** [review]

**Comment by qwertyquerty** [comment]
Still not 100% that letting the website be an intermediary between the game servers for almost no reason is a good idea

**Comment by ike709** [comment]
TODO: Have the cache be like 2 minutes, and just block double payments if poor timing happens.

**Pull request closed without merging**